assets/src/blocks/Spreadsheet/SpreadsheetEditor.js

```diff
@@ -0,0 +1,107 @@
+import {useState} from '@wordpress/element';
+import {InspectorControls} from '@wordpress/block-editor';
+import ColorPaletteControl from '../../block-editor/ColorPaletteControl/ColorPaletteControl';
+import {SpreadsheetFrontend} from './SpreadsheetFrontend';
+import {debounce} from '@wordpress/compose';
+import {TextControl, PanelBody} from '@wordpress/components';
+
+const {__} = wp.i18n;
+
+const colors = [
+  {name: 'blue', color: '#167f82'},
+  {name: 'green', color: '#1f4912'},
+  {name: 'grey', color: '#45494c'},
+  {name: 'gp-green', color: '#198700'},
+];
+
+export const SpreadsheetEditor = ({
+  attributes,
+  setAttributes,
+  isSelected,
+}) => {
+  const [invalidSheetId, setInvalidSheetId] = useState(false);
+  const [url, setUrl] = useState(attributes.url);
+
+  const debounceUrl = debounce(newUrl => {
+    setAttributes({url: newUrl});
+  }, 300);
+
+  const toColorName = code => colors.find(color => color.color === code)?.name || 'grey';
+
+  const toColorCode = name => colors.find(color => color.name === name)?.color || '#ececec';
```

Comment:
Same for colors

Reply:
This color is also part of the p4 colors, the above code is just the name associated to the color